BinaryObjectScanner/FileType/RAR.cs

```diff
@@ -51,6 +52,12 @@ public bool Extract(Stream? stream, string file, string outDir, bool lookForHead
                 if (!rarFile.IsComplete)
                     return false;
 
+                // Explained in https://github.com/adamhathcock/sharpcompress/pull/661. in order to determine whether  
+                // a RAR or 7Z archive is solid or not, you must check the second file in the archive, as the first 
+                // file is always marked non-solid even for solid archives. This iteration is necessary since things
+                // like directories aren't marked solid either.
+                bool firstFile = true;
```

Comment:
This entire block is strange in both implementations. The archive itself should have an `IsSolid` property.